src/provider/cloud-element-templates/Validator.js

```diff
@@ -70,7 +72,26 @@ export class Validator extends BaseValidator {
       }
     }
 
-    // (4) JSON schema compliance
+    // (4) elementType validation
+    if (template.elementType && template.appliesTo) {
+
+      const elementType = template.elementType.value,
+            appliesTo = template.appliesTo;
+
+      // (3.1) template can be applied to elementType
+      if (!appliesTo.find(type => isType(elementType, type))) {
+        return this._logError(`template does not apply to requested element type <${ elementType }>`, template);
+      }
+
+      // (3.2) template only applies to same type of element
+      for (const sourceType of appliesTo) {
+        if (!canMorph(elementType, sourceType)) {
+          return this._logError(`can not morph <${sourceType}> into <${elementType}>`, template);
+        }
+      }
+    }
+
+    // (5) JSON schema compliance
```

Comment:
I wonder whether we should validate first against the schema, and then do the more specific type check. This way some generic validations would happen before which we can rely on (e.g. whether `template.appliesTo` is available or not).

No hard requirement though.

Reply:
I think it's fine 👍